doc/arch/adr-007-pa-dep-client-app.md

```diff
@@ -0,0 +1,122 @@
+# Integrating a New Client App for PA DEP
+
+## Context
+
+The Pennsylvania Department of Environmental Protection (PA DEP) has been using desktop MapShed and MMW for their workflows for some time.
+This workflow involves running the MapShed model in MMW for a HUC-12,
+and also for an area of interest within the HUC-12,
+and copying the output tables into specific parts of the [MMW BMP Spreadsheet Tool][bmp-tool].
+This is cumbersome and error prone,
+and would be much better be done in an automated fashion,
+where when users download the spreadsheet it is pre-populated with the right values in the right places.
+
+Since this requires having two effective areas of interest at once,
+this workflow does not fit neatly in MMW,
+which always expects a single area of interest.
+Rather than change MMW to accomodate this unusual and rare use case,
+it would be much simpler to create a new client app that is focused on this workflow.
+It will use the MMW API for all calculations.
+
+Given that we'll have a new client app,
+this is an opportunity to introduce a more modern tech stack on the front-end.
+This was recently done for [a similar project][beekeepers],
+where a new front-end stack was added within a Django app to sandbox it from the rest of the project.
+This document aims to discuss the pros and cons of different arrangements for the new client app,
+and pick one that satisfies all project constraints.
+
+The one major directive from the clients is that,
+from a user's perspective,
+the site should be deeply integrated with MMW.
+That is to say,
+it should use the same colors and branding,
+and not feel like its own project.
+
+From the development perspective,
+we should aim for using the latest front-end stacks,
+clean communication with the back-end API,
+and having a fast deployment pipeline.
+
+### Repository: Same vs Different
+
+Advantages of having the **same repository**:
+
+  * Familiarity, as it was done in the [Beekeepers][beekeepers] project
+  * All issues can be in one repo, simplifying project management
+  * Tasks that affect both front-end and back-end can be tracked in one commit or PR
+  * Simpler deployment, as front-end and back-end will always be in sync
+  * Styles can be easily reused between the two apps
+  * Easier to simulate production environments in development,
```

Comment:
Just wanted to add that there is precedent for adjusting the client application's backend configuration such that it targets staging vs. a local development environment. This works for a lot of cases, but falls apart a bit in the event that a set of changes span the frontend and backend.